src/shared/crypto/bitcoin/bitcoin.utils.ts

```diff
@@ -61,36 +59,48 @@ export function decodeBitcoinTx(tx: string) {
   return btc.RawTx.decode(hexToBytes(tx));
 }
 
-const concat = P.concatBytes;
-
-function formatKey(hashed: Uint8Array, prefix: number[]): string {
-  return btc.base58check.encode(concat(Uint8Array.from(prefix), hashed));
-}
-
-function getAddressFromWshOutScript(script: Uint8Array, network: BtcSignerNetwork) {
-  return btc.programToWitness(0, script.slice(2), network);
-}
-
-function getAddressFromWpkhOutScript(script: Uint8Array, network: BtcSignerNetwork) {
-  return btc.programToWitness(0, script.slice(2), network);
-}
-
-function getAddressFromTrOutScript(script: Uint8Array, network: BtcSignerNetwork) {
-  return btc.programToWitness(1, script.slice(2), network);
-}
-
-export function getAddressFromOutScript(script: Uint8Array, network: BitcoinNetworkModes) {
-  const outScript = btc.OutScript.decode(script);
-  // This appears to be undefined at times?
-  const bitcoinNetwork = getBtcSignerLibNetworkConfigByMode(network);
-
-  if (outScript.type === 'wsh') return getAddressFromWshOutScript(script, bitcoinNetwork);
-  else if (outScript.type === 'wpkh') return getAddressFromWpkhOutScript(script, bitcoinNetwork);
-  else if (outScript.type === 'tr') return getAddressFromTrOutScript(script, bitcoinNetwork);
-  else if (outScript.type === 'pkh') return formatKey(script, [bitcoinNetwork?.pubKeyHash]);
-  else if (outScript.type === 'sh') return formatKey(script, [bitcoinNetwork?.scriptHash]);
-  logger.error(`Unknown address type=${outScript.type}`);
-  return '';
+// Outscript types (btc-signer lib):
+// type OutPKType = { type: 'pk'; pubkey: btc.Bytes }; - Public Key (P2PK)
+// type OutPKHType = { type: 'pkh'; hash: btc.Bytes }; - Publick Key Hash (P2PKH)
+// type OutSHType = { type: 'sh'; hash: btc.Bytes }; - Script Hash (P2SH)
+// type OutWSHType = { type: 'wsh'; hash: btc.Bytes }; - Witness Script Hash (P2WSH)
+// type OutWPKHType = { type: 'wpkh'; hash: btc.Bytes }; - Witness Public Key Hash (P2WPKH)
+// type OutMSType = { type: 'ms'; pubkeys: btc.Bytes[]; m: number }; - Multisig (P2MS)
+// type OutTRType = { type: 'tr'; pubkey: btc.Bytes }; - Taproot (P2TR)
+// type OutTRNSType = { type: 'tr_ns'; pubkeys: btc.Bytes[] }; - Taproot N-of-N multisig (P2TR_NS)
+// type OutTRMSType = { type: 'tr_ms'; pubkeys: btc.Bytes[]; m: number }; - Taproot M-of-N multisig (P2TR_MS)
+// type OutUnknownType = { type: 'unknown'; script: btc.Bytes }; - Unknown output type
+
+export function getAddressFromOutScript(script: Uint8Array, bitcoinNetwork: BtcSignerNetwork) {
+  const outputScript = btc.OutScript.decode(script);
+
+  if (outputScript.type === 'pk' || outputScript.type === 'tr') {
+    return btc.Address(bitcoinNetwork).encode({
+      type: outputScript.type,
+      pubkey: outputScript.pubkey,
+    });
+  } else if (outputScript.type === 'ms' || outputScript.type === 'tr_ms') {
```

Comment:
always feel like just using `if` is better than `else if` in such code. but up to you here

Reply:
Ok, yep, I'll refactor it. 👍